src/middle/Stmt.ml

```diff
@@ -208,6 +224,8 @@ module Helpers = struct
         in
         mk_for_iteratee rows (fun e -> for_each bodyfn e smeta) iteratee smeta
     | UArray _ -> mk_for_iteratee (len iteratee) bodyfn iteratee smeta
+    (* TUPLE MAYBE It's not clear that this should ever be called: *)
+    | UTuple ts -> for_each_tuple (const bodyfn) iteratee ts smeta
```

Comment:
This is never called. If the tuple is not homogeneous then `bodyfn` can't type-check. Let's not add MIR features unless there's a compelling use case.